src/upgrade-dependencies.ts

```diff
@@ -0,0 +1,368 @@
+import { Component } from './component';
+import { GithubWorkflow } from './github';
+import { NodeProject } from './node-project';
+import { Task } from './tasks';
+
+function context(value: string) {
+  return `\${{ ${value} }}`;
+}
+
+function setOutput(name: string, value: string) {
+  return `echo "::set-output name=${name}::${value}"`;
+}
+
+const RUNNER_TEMP = context('runner.temp');
+const DEFAULT_TOKEN = context('secrets.GITHUB_TOKEN');
+const REPO = context('github.repository');
+const RUN_ID = context('github.run_id');
+const RUN_URL = `https://github.com/${REPO}/actions/runs/${RUN_ID}`;
+const UBUNTU_LATEST = 'ubuntu-latest';
+
+/**
+ * Options for `UpgradeDependencies.workflowOptions`.
+ */
+export interface UpgradeDependenciesWorkflowOptions {
+
+  /**
+   * Schedule to run on.
+   *
+   * @default UpgradeDependenciesSchedule.DAILY
+   */
+  readonly schedule?: UpgradeDependenciesSchedule;
+
+  /**
+   * Which secret to use when creating the PR.
+   *
+   * @default - default github token.
+   */
+  readonly secret?: string;
+
+  /**
+   * Labels to apply on the PR.
+   *
+   * @default - no labels.
+   */
+  readonly labels?: string[];
+
+  /**
+   * Execute 'build' after the upgrade.
+   *
+   * @default true
+   */
+  readonly rebuild?: boolean;
+
+}
+
+/**
+ * Options for `UpgradeDependencies`.
+ */
+export interface UpgradeDependenciesOptions {
+
+  /**
+   * List of package names to exclude during the upgrade.
+   *
+   * @default - Nothing is excluded.
+   */
+  readonly exclude?: string[];
+
+  /**
+   * List of package names to include during the upgrade.
+   *
+   * @default - Everything is included.
+   */
+  readonly include?: string[];
+
+  /**
+   * Include a github workflow for creating PR's that upgrades the
+   * required dependencies.
+   *
+   * @default - true for root projects, false for sub-projects.
+   */
+  readonly workflow?: boolean;
+
+  /**
+   * Options for the github workflow. Only applies if `workflow` is true.
+   *
+   * @default - default options.
+   */
+  readonly workflowOptions?: UpgradeDependenciesWorkflowOptions;
+
+  /**
+   * The name of the task that will be created.
+   * This will also be the workflow name.
+   *
+   * @default 'upgrade-dependencies'.
```

Comment:
```suggestion
   * @default "upgrade-dependencies"
```

Defaults should be JSON